src/watertap_contrib/seto/costing/units/chemical_softening_0D.py

```diff
@@ -0,0 +1,833 @@
+
+    blk.lime_feed_system_capital_cost = Var(
+        initialize=1e6,
+        doc="Capital cost of lime feed system",
+        domain=NonNegativeReals,
+        units=blk.costing_package.base_currency,
+    )
+
+    blk.admin_capital_cost = Var(
+        initialize=1e6,
+        doc="Capital cost of administrative, laboratory, and maintenance building",
+        domain=NonNegativeReals,
+        units=blk.costing_package.base_currency,
+    )
+    # endregion
+
+    # region Operating cost components
+
+    blk.mix_tank_op_cost = Var(
+        initialize=1e4,
+        doc="Operational cost of mixing tank - energy consumption and labor",
+        domain=NonNegativeReals,
+        units=blk.costing_package.base_currency,
+    )
+
+    blk.floc_tank_op_cost = Var(
+        initialize=1e4,
+        doc="Operational cost of flocculation tank - energy consumption and labor",
+        domain=NonNegativeReals,
+        units=blk.costing_package.base_currency,
+    )
+
+    blk.sed_basin_op_cost = Var(
+        initialize=1e4,
+        doc="Operational cost of sedimentation basin - energy consumption, labor, and maintenance",
+        domain=NonNegativeReals,
+        units=blk.costing_package.base_currency,
+    )
+
+    blk.recarb_basin_op_cost = Var(
+        initialize=1e4,
+        doc="Operationg cost of recarbonation basin - energy consumption, labor, and maintenance",
+        domain=NonNegativeReals,
+        units=blk.costing_package.base_currency,
+    )
+
+    blk.lime_feed_op_cost = Var(
+        initialize=1e4,
+        doc="Operation cost of lime feed system - energy consumption, labor, and maintenance",
+        domain=NonNegativeReals,
+        units=blk.costing_package.base_currency,
+    )
+
+    blk.soda_ash_add_op_cost = Var(
+        initialize=1e4,
+        doc="Operation cost of adding soda ash - chemical addition cost",
+        domain=NonNegativeReals,
+        units=blk.costing_package.base_currency,
+    )
+
+    if blk.unit_model.config.silica_removal:
+        blk.mg_add_op_cost = Var(
+            initialize=1e4,
+            doc="Operational cost of adding Mg - chemical addition cost",
+            domain=NonNegativeReals,
+            units=blk.costing_package.base_currency,
+        )
+
+    blk.lime_sludge_mngt_op_cost = Var(
+        initialize=1e5,
+        doc="Operational cost of lime sludge management",
+        domain=NonNegativeReals,
+        units=blk.costing_package.base_currency,
+    )
+
+    blk.admin_op_cost = Var(
+        initialize=1e5,
+        doc="Operational cost of administrative, laboratory and maintenance building",
+        domain=NonNegativeReals,
+        units=blk.costing_package.base_currency,
+    )
+
+    # endregion
+
+    # region Capital cost component constraints
+
+    # Mixing tank
+    if value(blk.unit_model.vel_gradient_mix) <= 300:
+        blk.mix_tank_capital_cost_constraint = Constraint(
+            expr=blk.mix_tank_capital_cost
+            == (
+                0.0002
```

Comment:
We don't want hardcoded values in WaterTAP models typically. These should be defined in ``build_chem_softening_cost_param_block`` and can be called something like "tank_cap_cost_A_coeff". And should probably be ``Params`` rather than ``Vars``.

Reply:
It'd be interesting to see a plot of what the cost is vs `vel_gradient_mix`. Perhaps a smooth surrogate could also be created and used to replace these conditionals.

Reply:
That's true. Having them as Vars or mutable Params would also eliminate the need for all these conditionals since the user could then just fix them to whatever is appropriate for the velocity gradient they are using.

I would also like to see a plot of the different powers generated from each regression for velocity gradient. I am not convinced using one vs another would result in significant differences to LCOW or energy consumption, so leaving the option might add unnecessary complexity to a otherwise pretty simple model.